docs/plugins/development/creating-non-text-analysis-plugins.asciidoc

```diff
@@ -0,0 +1,93 @@
+that describes the plugin.
+
+Note that only JAR files at the root of the plugin are added to the classpath
+for the plugin. If you need other resources, package them into a resources JAR.
+
+[discrete]
+==== Example plugins
+
+The {es} repository contains {es-repo}tree/main/plugins/examples[examples of plugins]. Some of these include:
+
+* a plugin with {es-repo}tree/main/plugins/examples/custom-settings[custom settings]
+* adding {es-repo}tree/main/plugins/examples/rest-handler[custom rest endpoints]
+* adding a {es-repo}tree/main/plugins/examples/rescore[custom rescorer]
+* a script {es-repo}tree/main/plugins/examples/script-expert-scoring[implemented in Java]
+
+These examples provide the bare bones needed to get started. For more
+information about how to write a plugin, we recommend looking at the plugins
+listed in this documentation for inspiration.
+
+[discrete]
+==== Testing your plugin
+
+When testing a Java plugin, it will only be auto-loaded if it is in the
+`plugins/` directory. Use `bin/elasticsearch-plugin install file:///path/to/your/plugin`
+to install your plugin for testing.
+
+You may also load your plugin within the test framework for integration tests.
+Read more in {ref}/integration-tests.html#changing-node-configuration[Changing Node Configuration].
+
+[discrete]
+[[plugin-authors-jsm]]
+==== Java Security permissions
+
+Some plugins may need additional security permissions. A plugin can include
+the optional `plugin-security.policy` file containing `grant` statements for
+additional permissions. Any additional permissions will be displayed to the user
+with a large warning, and they will have to confirm them when installing the
+plugin interactively. So if possible, it is best to avoid requesting any
+spurious permissions!
+
+If you are using the Elasticsearch Gradle build system, place this file in
+`src/main/plugin-metadata` and it will be applied during unit tests as well.
+
+Keep in mind that the Java security model is stack-based, and the additional
+permissions will only be granted to the jars in your plugin, so you will have
+write proper security code around operations requiring elevated privileges.
+It is recommended to add a check to prevent unprivileged code (such as scripts)
+from gaining escalated permissions. For example:
+
+[source,java]
+--------------------------------------------------
+// ES permission you should check before doPrivileged() blocks
+import org.elasticsearch.SpecialPermission;
+
+SecurityManager sm = System.getSecurityManager();
+if (sm != null) {
+  // unprivileged code such as scripts do not have SpecialPermission
+  sm.checkPermission(new SpecialPermission());
+}
+AccessController.doPrivileged(
+  // sensitive operation
+);
+--------------------------------------------------
+
+See https://www.oracle.com/technetwork/java/seccodeguide-139067.html[Secure Coding Guidelines for Java SE]
+for more information.
+
```

Comment:
We have a gradle plugin to build these plugins as well, can we mention it similar to how it is mentioned for stable plugins?

Reply:
I'd like to follow up with another PR to update the non-text analysis plugin doc page. We can make it similar to the text analysis plugins docs: with a step by step and a hello world example. We can do that after the 8.7 release though.

Reply:
@rjernst I've updated the docs to use 'stable' and 'classic' terminology. Let me know what you think.